Xamarin.ExposureNotification/ExposureNotification.android.cs

```diff
@@ -37,14 +42,76 @@ static async Task<ExposureConfiguration> GetConfigurationAsync()
 				.Build();
 		}
 
+		const int requestCodeStartExposureNotification = 1111;
+		//public static final int REQUEST_CODE_GET_TEMP_EXPOSURE_KEY_HISTORY = 2222;
+
+		static TaskCompletionSource<object> tcsResolveConnection;
+
+		public static void OnActivityResult(int requestCode, Result resultCode, global::Android.Content.Intent data)
+		{
+			if (resultCode == Result.Ok)
+				tcsResolveConnection?.TrySetResult(null);
+			else
+				tcsResolveConnection.TrySetException(new AccessDeniedException("Failed to resolve enabing Exposure Notifications"));
+		}
+
+		static async Task ResolveApi(Func<Task> apiCall)
+		{
+			try
+			{
+				await apiCall();
+			}
+			catch (ApiException apiEx)
+			{
+				if (apiEx.StatusCode == 6) // Resolution required
+				{
+					tcsResolveConnection = new TaskCompletionSource<object>();
+
+					var statusProperty = apiEx.GetType().GetProperty("MStatus", System.Reflection.BindingFlags.Instance | System.Reflection.BindingFlags.NonPublic);
+					var val = (Java.Lang.Object)statusProperty.GetValue(apiEx);
+					var statuses = val.JavaCast<Statuses>();
+
+					statuses.StartResolutionForResult(Essentials.Platform.CurrentActivity, requestCodeStartExposureNotification);
+
+					await tcsResolveConnection.Task;
+
+					await apiCall();
+				}
+			}
+		}
+
 		static Task PlatformStart()
-			=> Instance.StartAsync();
+			=> ResolveApi(() => Instance.StartAsync());
 
 		static Task PlatformStop()
-			=> Instance.StopAsync();
+			=> ResolveApi(() => Instance.StopAsync());
 
 		static async Task<bool> PlatformIsEnabled()
-			=> await Instance.IsEnabledAsync();
+		{
+			try
+			{
+				return await Instance.IsEnabledAsync();
+			}
+			catch (ApiException apiEx)
+			{
+				if (apiEx.StatusCode == 6) // Resolution required
+				{
+					tcsResolveConnection = new TaskCompletionSource<object>();
+
+					var statusProperty = apiEx.GetType().GetProperty("MStatus", System.Reflection.BindingFlags.Instance | System.Reflection.BindingFlags.NonPublic);
+					var val = (Java.Lang.Object)statusProperty.GetValue(apiEx);
+					var statuses = val.JavaCast<Statuses>();
+
+					statuses.StartResolutionForResult(Essentials.Platform.CurrentActivity, requestCodeStartExposureNotification);
+
+					await tcsResolveConnection.Task;
+
+					return await Instance.IsEnabledAsync();
+				}
+			}
```

Comment:
This looks to be duplicate code. Should it be `ResolveApi`?

Reply:
Yeah it's just weird because one returns a type and one is a void.